code/modules/mob/living/carbon/xenomorph/hive_datum.dm

```diff
@@ -433,36 +499,51 @@ to_chat will check for valid clients itself already so no need to double check f
 
 
 /datum/hive_status/normal/proc/can_spawn_larva(mob/xeno_candidate)
+	if(!xeno_candidate?.client)
+		return FALSE
+
 	if(!stored_larva)
 		to_chat(xeno_candidate, "<span class='warning'>There are no burrowed larvas.</span>")
 		return FALSE
 
-	if(!living_xeno_queen?.ovipositor)
-		to_chat(xeno_candidate, "<span class='warning'>There are no mothers with an ovipositor deployed.</span>")
+	var/list/possible_mothers = list()
+	if(living_xeno_queen?.is_burrowed_larva_host())
```

Comment:
i dont want to look at is_burrowed_larva_host because i know its going to be awful

Reply:
You've complained about being ignored before.
This kind of useless comment might be one of the reasons.
You've made 3 comments saying how awful it is, and given zero suggestions about it.
If you want me to listen to you and take you seriously then please at least suggest a solution.
Make your comments have a purpose, or don't make them at all.